src/multi_translator.cc

```diff
@@ -73,83 +81,90 @@ BiltransToken MultiTranslator::getFullToken(UString source) {
+	{
+		target += '@';
+		target.append(source);
 	}
 	token = parseBiltransToken(source + "/"_u + target);
 	return token;
-
 }
 
 BiltransToken MultiTranslator::getTrimmedToken(UString source)
 {
 	BiltransToken ttoken;
 	BiltransToken ftoken;
 
-	if (source[0] == '*') {
+	if (source[0] == '*')
+	{
 		ttoken.sourceToken.lemma = source;
 		TaggerToken tmp;
 		tmp.lemma = source;
 		ttoken.targetTokens.push_back(tmp);
 		return ttoken;
 	}
 
-        /*---------------------------------------------*/
+	/*---------------------------------------------*/
 
-        // This code is to attempt to avoid memory leak problems when calling
-        // the bilingual.* methods in FSTProcessor. Unknown why we get the
-        // leaks in the first place...
+	// This code is to attempt to avoid memory leak problems when calling
+	// the bilingual.* methods in FSTProcessor. Unknown why we get the
+	// leaks in the first place...
 
-        UString fstr;
-        UString tstr;
+	UString fstr;
+	UString tstr;
 
-	if((f_cache.find(source) == f_cache.end()))
-        {
-	  f_cache[source] = bilingual.biltrans(source, false);
-        }
-	if((t_cache.find(source) == t_cache.end()))
-        {
-	  t_cache[source] = bilingual.biltransWithoutQueue(source, false);
-        }
+	if ((f_cache.find(source) == f_cache.end()))
+	{
+		f_cache[source] = bilingual.biltrans(source, false);
+	}
+	if ((t_cache.find(source) == t_cache.end()))
+	{
+		t_cache[source] = bilingual.biltransWithoutQueue(source, false);
+	}
 
-        fstr = f_cache[source];
-        tstr = t_cache[source];
+	fstr = f_cache[source];
+	tstr = t_cache[source];
 
-        /*---------------------------------------------*/
+	/*---------------------------------------------*/
 
-        if (fstr.empty()) {
-          fstr += '@';
-          fstr.append(source);
-        }
-        if (tstr.empty()) {
-          tstr += '@';
-          tstr.append(source);
-        }
+	if (fstr.empty())
+	{
+		fstr += '@';
+		fstr.append(source);
+	}
+	if (tstr.empty())
+	{
+		tstr += '@';
+		tstr.append(source);
+	}
 
 	ttoken = parseBiltransToken(source + "/"_u + tstr);
 	ftoken = parseBiltransToken(source + "/"_u + fstr);
 
-
-	if(this->trimmed) {
-		for(size_t i = 0; i < ftoken.targetTokens.size(); ++i ) {
-			if(ttoken.targetTokens[i].tags.size() <
-			   ftoken.targetTokens[i].tags.size()) {
+	if (this->trimmed)
+	{
+		for (size_t i = 0; i < ftoken.targetTokens.size(); ++i)
+		{
+			if (ttoken.targetTokens[i].tags.size() <
+				ftoken.targetTokens[i].tags.size())
+			{
 				ttoken.targetTokens[i].tags.push_back("*"_u);
 			}
 		}
 	}
 
 	vector<UString> newTags;
 	//bool sourceTrimmed = false;
-	for(size_t i = 0; i < ttoken.sourceToken.tags.size(); ++i) {
+	for (size_t i = 0; i < ttoken.sourceToken.tags.size(); ++i)
+	{
 		UString tag = ttoken.sourceToken.tags[i];
-		if (find(ttoken.targetTokens[0].tags, tag) ==
-			find(ftoken.targetTokens[0].tags, tag)) {
+		if ((int)i == find(ftoken.targetTokens[0].tags, tag))
+		{
```

Comment:
scanning through, it looks like this is the only non-formatting change

Reply:
Thus proving my point. Line 44's for() loop is also changed to omit the first element.

Reply:
yep, totally missed that it was `i = 1` rather than `i = 0`.